bin/terraform-deploy

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+set -e
+
+if [ $# -eq 0 ]
+then
+  echo "No terraform workspace supplied, aborting."
+  exit 1
+fi
+
+TERRAFORM_WORKSPACE="$@"
+GIT_BRANCH=""
+
+case $@ in
+  "production") GIT_BRANCH="master"
+    ;;
+  "staging") GIT_BRANCH="develop"
+    ;;
+  "edge") GIT_BRANCH="edge"
+    ;;
+  "testing") GIT_BRANCH="testing"
+    ;;
+  *)
+    echo "Could not find an known environment with the name: $@"
+    exit 1
+    ;;
+esac
+
+echo "Deploying…"
+echo "* Terraform workspace: $@"
+echo "* Git branch: $GIT_BRANCH"
+
+read -p "Continue (y/n)?" CONT
+if [ "$CONT" = "y" ]; then
+  echo
+else
+  exit 1
+fi
+
+echo "-> Checking out terraform workspace $@ …"
+terraform workspace select $@
+
+echo "-> Fetching latest git changes …"
+git fetch
+git pull origin $GIT_BRANCH
+
+echo "-> Checking out branch $GIT_BRANCH …"
+git checkout $GIT_BRANCH
+
+echo "-> Applying Terraform command on the $@ workspace …"
+terraform apply -var-file="workspace-variables/$@.tfvars"
```

Comment:
`shellcheck` isnt happy with a lot of this
```
bob@roti:~/git/dxw/teacher-vacancy-service ☃ >shellcheck bin/terraform-deploy

In bin/terraform-deploy line 10:
TERRAFORM_WORKSPACE="$@"
^-- SC2034: TERRAFORM_WORKSPACE appears unused. Verify it or export it.
                    ^-- SC2124: Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.


In bin/terraform-deploy line 23:
    echo "Could not find an known environment with the name: $@"
                                                             ^-- SC2145: Argument mixes string and array. Use * or separate argument.


In bin/terraform-deploy line 29:
echo "* Terraform workspace: $@"
                             ^-- SC2145: Argument mixes string and array. Use * or separate argument.


In bin/terraform-deploy line 32:
read -p "Continue (y/n)?" CONT
^-- SC2162: read without -r will mangle backslashes.


In bin/terraform-deploy line 39:
echo "-> Checking out terraform workspace $@ …"
                                          ^-- SC2145: Argument mixes string and array. Use * or separate argument.


In bin/terraform-deploy line 40:
terraform workspace select $@
                           ^-- SC2068: Double quote array expansions to avoid re-splitting elements.


In bin/terraform-deploy line 49:
echo "-> Applying Terraform command on the $@ workspace …"
                                           ^-- SC2145: Argument mixes string and array. Use * or separate argument.


In bin/terraform-deploy line 50:
terraform apply -var-file="workspace-variables/$@.tfvars"
                                               ^-- SC2145: Argument mixes string and array. Use * or separate argument.
```